src/Hedgehog/Property.fs

```diff
@@ -56,15 +56,18 @@ module Property =
             failure
 
     let counterexample (msg : unit -> string) : Property<unit> =
-        Gen.constant (Journal.singleton msg, Success ()) |> ofGen
+        (Journal.singleton msg, Success ()) |> Gen.constant |> ofGen
 
     let private mapGen
             (f : Gen<Journal * Outcome<'a>> -> Gen<Journal * Outcome<'b>>)
-            (x : Property<'a>) : Property<'b> =
-        toGen x |> f |> ofGen
+            (p : Property<'a>) : Property<'b> =
+        p |> toGen |> f |> ofGen
 
     let map (f : 'a -> 'b) (x : Property<'a>) : Property<'b> =
-        (mapGen << GenTuple.mapSnd << Outcome.map) f x
+        f |> Outcome.map |> GenTuple.mapSnd |> mapGen <| x
+
+    let set (a: 'a) (property : Property<'b>) : Property<'a> =
```

Comment:
Should this be `private`?

Reply:
If `Property<>` remains generic, then I think so, but since we are considering removing the generics, sure.  Done.